api/admin/controllers/site.js

```diff
@@ -69,4 +73,25 @@ module.exports = wrapHandlers({
 
     return res.json(serializeNew(site, true));
   },
+
+  destroy: async (req, res) => {
+    let site;
+    const { id } = req.params;
+
+    try {
+      site = await fetchSiteById(id);
+      await S3SiteRemover.removeSite(site);
+      await S3SiteRemover.removeInfrastructure(site);
+
+      if (Features.enabled(Features.Flags.FEATURE_PROXY_EDGE_DYNAMO)) {
+        await ProxyDataSync.removeSite(site);
+      }
+
+      await site.destroy();
```

Comment:
Do all of the steps here need to happen serially, or could they happen concurrently?

Reply:
also, this logic is identical to the logic in the user client, perhaps it should be refactored to a common service...

Reply:
👍 Pulled out a `SiteDestroy` service